### ISSUE TITLE
AlternativeOutputFormats generates RSS links when disableKinds=["RSS"] is set

### ISSUE THREAD

**Issue by gbirke** [gohugoio/hugo#4829]
I've set `disableKinds = [ "taxonomy", "taxonomyTerm", "RSS", "sitemap" ]`

I followed the instructions in https://github.com/gohugoio/hugo/issues/4427#issuecomment-366760534 and used 

```
{{ range .AlternativeOutputFormats -}}
    <link rel="{{ .Rel }}" type="{{ .MediaType.Type }}" href="{{ .Permalink | safeURL }}">
{{ end -}}
```

in my template. I'd have expected `.AlternativeOutputFormats` to be empty, but the following RSS link was still generated:

```
<link rel="alternate" type="application/rss" href="http://localhost:1313/index.xml">
```

Is this a bug or do I have to set something else to avoid generating RSS links?

I'm using Hugo 0.41 on Mac OS X

**Comment by gbirke** [comment]
The bug is still present in Hug 0.54.0 on Mac OS X.

**Comment by gbirke** [comment]
The behavior is still present in Hugo 0.55, tested both on macOS and Linux, so it's not platform-dependent.

But there is a new behavior - the deprecated `{{ if .RSSLink }}` syntax now also generates an RSS link when RSS is disables via `disableKinds`.

I've experimented some more and found the following patten: `disableKinds` is independent from the `outputs` settings. `disableKinds` suppresses the generation of the files, but does not affect the generated outputs, so  RSS is still part of `.AlternativeOutputFormats`. If I set `[outputs]` to `home=["HTML"]`, neither the RSS *link* nor the RSS *file* are generated. I think the opposite should also be true: when `AlternativeOutputFormats` contains RSS, the `.AlternativeOutputFormats` variable should not contain RSS, even if set explicitly in `[outputs]`

**Comment by gbirke** [comment]
The issue still persists with Hugo 0.57:

`config.toml` contains
```
disableKinds = [ "taxonomy", "taxonomyTerm", "RSS", "sitemap" ]
```

`head.html` contains
```
{{ with .OutputFormats.Get "rss" -}}
      {{ printf `<link rel="%s" type="%s" href="%s" title="%s" />` .Rel .MediaType.Type .Permalink $.Site.Title | safeHTML }}
{{ end -}}
````

Expected result: Link is not printed

Actual result: Link is printed.

**Comment by johnelliott** [comment]
I think I am running in to this too.

**Comment by Teknomancer** [comment]
This issue still persists with Hugo 0.62.

Is there a "clean" / suggested way of handling this?

**Comment by bep** [comment]
I will fix this in #6738 -- but you should really look into how to configure what output formats to produce.
